swift/lib/dependabot/swift/requirement.rb

```diff
@@ -15,12 +15,13 @@ class Requirement < Dependabot::Requirement
       # always contains a single element.
       sig { override.params(requirement_string: T.nilable(String)).returns(T::Array[Requirement]) }
       def self.requirements_array(requirement_string)
-        [new(requirement_string)]
+        [new([T.must(requirement_string)])]
       end
 
       # Patches Gem::Requirement to make it accept requirement strings like
       # "~> 4.2.5, >= 4.2.5.1" without first needing to split them.
-      def initialize(*requirements)
+      sig { params(requirements: T::Array[String]).void }
```

Comment:
Sorbet recommended refactoring this way as support for splat arguments is minimal. https://sorbet.org/docs/error-reference#7019